"Failed to load the assembly" in Rewrite.exe when trying to build

### Description

Build "fails" with the following error. It actually seems to continue running the tests, though when trying to use the resulting `OpenTK.dll` I get a `BindingsNotRewrittenException`.

```
Target AfterBuild:
    Executing: mono bin/Release/../../../Generator.Rewrite/bin/Release/Rewrite.exe --assembly bin/Release/OpenTK.dll --signing-key ../../OpenTK.snk
    Failed to load the assembly. It may already have been rewritten, and the debug symbols no longer match.
    System.InvalidOperationException: Operation is not valid due to the current state of the object.
        at Mono.Cecil.ModuleDefinition.ReadSymbols (Mono.Cecil.Cil.ISymbolReader reader) [0x0002f] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at Mono.Cecil.ModuleReader.ReadSymbols (Mono.Cecil.ModuleDefinition module, Mono.Cecil.ReaderParameters parameters) [0x0004a] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at Mono.Cecil.ModuleReader.CreateModule (Mono.Cecil.PE.Image image, Mono.Cecil.ReaderParameters parameters) [0x00081] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at Mono.Cecil.ModuleDefinition.ReadModule (Mono.Disposable`1[T] stream, System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x0000d] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at Mono.Cecil.ModuleDefinition.ReadModule (System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x0006c] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at Mono.Cecil.AssemblyDefinition.ReadAssembly (System.String fileName, Mono.Cecil.ReaderParameters parameters) [0x00000] in <2e24b151dd8041d8b7d36a855bb53c66>:0
        at OpenTK.Rewrite.Program.Rewrite () [0x00099] in <093950cc61144551b17c56e82fd446a2>:0
```

### Repro steps

Run `./build.sh`

### Related information 

- **Operating system:** Arch Linux
- **What runtime:** Mono `5.0.0`

For another project I compile using `dotnet`, I have to specify the `FrameworkPathOverride` enviroment variable, in my case to `"/lib/mono/4.5/"`. It's actually a [tracked issue on dotnet](https://github.com/dotnet/sdk/issues/335#issuecomment-322137207) right now. Not sure if this is related to this or not. I should give it a try. .... huh.

Using `4.5` makes them resolve properly, but:
```
mono bin/Release/../../../Generator.Rewrite/bin/Release/Rewrite.exe --assembly bin/Release/OpenTK.dll --signing-key ../../OpenTK.snk
  Debugging symbols for target assembly not found.
  Please make sure that debugging symbols are being generated.
```
Using `2.0-api`, compiling fails:
```
  /usr/lib/mono/4.5/mcs.exe /noconfig /nowarn:1701,1702 /nostdlib+ /platform:anycpu32bitpreferred /warn:4 /doc:bin/Release/Bind.xml /baseaddress:285212672 /define:TRACE /highentropyva+ /reference:/lib/mono/2.0-api/mscorlib.dll /reference:/lib/mono/2.0-api/System.Core.dll /reference:/lib/mono/2.0-api/System.dll /reference:/lib/mono/2.0-api/System.Xml.dll /reference:/lib/mono/2.0-api/System.Xml.Linq.dll /debug- /filealign:4096 /keyfile:../../OpenTK.snk /optimize+ /deterministic+ /out:obj/Release/Bind.exe /ruleset:../../stylecop.ruleset /subsystemversion:6.00 /target:exe /utf8output /analyzer:../../packages/StyleCop.Analyzers/analyzers/dotnet/cs/StyleCop.Analyzers.CodeFixes.dll /analyzer:../../packages/StyleCop.Analyzers/analyzers/dotnet/cs/StyleCop.Analyzers.dll CSharpSpecWriter.cs FuncProcessor.cs GL2/GL4Generator.cs IBind.cs Main.cs DocProcessor.cs Settings.cs BindStreamWriter.cs EnumProcessor.cs Utilities.cs ISpecReader.cs Properties/AssemblyInfo.cs CL/CLGenerator.cs GL2/Generator.cs Structures/Type.cs Structures/Function.cs Structures/Constant.cs Structures/Enum.cs Structures/Delegate.cs Structures/Parameter.cs Structures/FlowDirection.cs ES/ESGenerator.cs XmlSpecReader.cs ES/ES2Generator.cs ES/ES3Generator.cs GL2/GL2Generator.cs Structures/Documentation.cs ES/ES31Generator.cs "/tmp/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs"
/tmp/.NETFramework,Version=v4.6.1.AssemblyAttributes.cs(4,46): error CS0234: The type or namespace name `TargetFrameworkAttribute' does not exist in the namespace `System.Runtime.Versioning'. Are you missing an assembly reference? [/home/copygirl/projects/opentk/src/Generator.Bind/Generator.Bind.csproj]
CSharpSpecWriter.cs(730,34): error CS1502: The best overloaded method match for `string.Join(string, string[])' has some invalid arguments [/home/copygirl/projects/opentk/src/Generator.Bind/Generator.Bind.csproj]
  /lib/mono/2.0-api/mscorlib.dll (Location of the symbol related to previous error)
CSharpSpecWriter.cs(730,45): error CS1503: Argument `#2' cannot convert `System.Collections.Generic.List<string>' expression to type `string[]' [/home/copygirl/projects/opentk/src/Generator.Bind/Generator.Bind.csproj]
```
Hope this helps cracking down on the issue somehow.

--- a/src/Generator.Rewrite/OpenTKAssemblyResolver.cs
+++ b/src/Generator.Rewrite/OpenTKAssemblyResolver.cs
@@ -93,11 +93,7 @@ namespace OpenTK.Rewrite
                         continue;
                     try
                     {
-                        var assembly = GetAssembly(file, parameters);
-                        if (assembly.Name.Version == name.Version || IsZero(name.Version))
-                        {
-                            return assembly;
-                        }
+                        return GetAssembly(file, parameters);
                     }
                     catch (System.BadImageFormatException)
                     {
@@ -248,11 +244,7 @@ namespace OpenTK.Rewrite
                 var file = GetAssemblyFile(reference, string.Empty, gac_path);
                 if (File.Exists(file))
                 {
-                    var assembly = GetAssembly(file, parameters);
-                    if (assembly.Name.Version == reference.Version || IsZero(reference.Version))
-                    {
-                        return assembly;
-                    }
+                    return GetAssembly(file, parameters);
                 }
             }
 
@@ -272,11 +264,7 @@ namespace OpenTK.Rewrite
                     var file = GetAssemblyFile(reference, prefixes[i], gac);
                     if (Directory.Exists(gac) && File.Exists(file))
                     {
-                        var assembly = GetAssembly(file, parameters);
-                        if (assembly.Name.Version == reference.Version || IsZero(reference.Version))
-                        {
-                            return assembly;
-                        }
+                        return GetAssembly(file, parameters);
                     }
                 }
             }

